src/__tests__/components/frontend-engine/frontend-engine.spec.tsx

```diff
@@ -111,76 +112,90 @@ describe("frontend-engine", () => {
 
-		expect(onChange).toBeCalledWith(expect.objectContaining({ [FIELD_ONE_ID]: "hello" }), true);
-	});
+			expect(onChange).toBeCalledWith(expect.objectContaining({ [FIELD_ONE_ID]: "hello" }), true);
+		});
 
-	it("should return the correct validity in the onChange prop when form is prefilled with valid values", async () => {
-		const onChange = jest.fn();
-		renderComponent(
-			{ onChange },
-			{
-				sections: {
-					section: {
-						uiType: "section",
-						children: {
-							[FIELD_ONE_ID]: {
-								uiType: "text-field",
-								label: FIELD_ONE_LABEL,
-								validation: [{ required: true }],
+		it("should return the correct validity when form is prefilled with valid values", async () => {
+			renderComponent(
+				{ onChange },
+				{
+					sections: {
+						section: {
+							uiType: "section",
+							children: {
+								[FIELD_ONE_ID]: {
+									uiType: "text-field",
+									label: FIELD_ONE_LABEL,
+									validation: [{ required: true }],
+								},
 							},
 						},
 					},
-				},
-				defaultValues: { [FIELD_ONE_ID]: "a" },
-			}
-		);
+					defaultValues: { [FIELD_ONE_ID]: "a" },
+				}
+			);
 
-		expect(onChange).toBeCalledWith(expect.objectContaining({ [FIELD_ONE_ID]: "a" }), true);
-	});
+			expect(onChange).toBeCalledWith(expect.objectContaining({ [FIELD_ONE_ID]: "a" }), true);
+		});
 
-	it("should return the correct validity in the onChange prop when form is prefilled with invalid values", async () => {
-		const onChange = jest.fn();
-		renderComponent(
-			{ onChange },
-			{
-				sections: {
-					section: {
-						uiType: "section",
-						children: {
-							[FIELD_ONE_ID]: {
-								uiType: "text-field",
-								label: FIELD_ONE_LABEL,
-								validation: [{ min: 5 }],
+		it("should return the correct validity when form is prefilled with invalid values", async () => {
+			renderComponent(
+				{ onChange },
+				{
+					sections: {
+						section: {
+							uiType: "section",
+							children: {
+								[FIELD_ONE_ID]: {
+									uiType: "text-field",
+									label: FIELD_ONE_LABEL,
+									validation: [{ min: 5 }],
+								},
 							},
 						},
 					},
-				},
-				defaultValues: { [FIELD_ONE_ID]: "a" },
-			}
-		);
+					defaultValues: { [FIELD_ONE_ID]: "a" },
+				}
+			);
 
-		expect(onChange).toBeCalledWith(expect.objectContaining({ [FIELD_ONE_ID]: "a" }), false);
-	});
+			expect(onChange).toBeCalledWith(expect.objectContaining({ [FIELD_ONE_ID]: "a" }), false);
+		});
 
-	it("should call onSubmit prop on submit", async () => {
-		const onSubmit = jest.fn();
-		renderComponent({
-			onSubmit,
+		it("should include form values of unregistered fields if stripUnknown is not true", () => {
+			renderComponent({ onChange }, { ...JSON_SCHEMA, defaultValues: { nonExistentField: "hello world" } });
+			fireEvent.change(getFieldOne(), { target: { value: "hello" } });
+
+			const finalOnChangeCall = onChange.mock.calls[onChange.mock.calls.length - 1][0];
```

Comment:
can use `onChange.mock.lastCall` or `toHaveBeenLastCalledWith` for this purpose

Reply:
thanks I have fixed them and learnt something new today 😄